src/test/java/net/minecraftforge/debug/gameplay/loot/GlobalLootModifiersTest.java

```diff
@@ -0,0 +1,212 @@
+import net.minecraft.inventory.EquipmentSlotType;
+import net.minecraft.inventory.Inventory;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.FurnaceRecipe;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.util.JSONUtils;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.storage.loot.LootContext;
+import net.minecraft.world.storage.loot.LootParameterSets;
+import net.minecraft.world.storage.loot.LootParameters;
+import net.minecraft.world.storage.loot.LootTable;
+import net.minecraft.world.storage.loot.conditions.ILootCondition;
+import net.minecraftforge.common.loot.LootModifier;
+import net.minecraftforge.common.loot.LootModifierManager;
+import net.minecraftforge.event.RegistryEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.Mod.EventBusSubscriber;
+import net.minecraftforge.registries.ObjectHolder;
+
+@Mod(GlobalLootModifiersTest.MODID)
+public class GlobalLootModifiersTest {
+	public static final String MODID = "global_loot_test";
+	public static final boolean ENABLE = true;
+	@ObjectHolder(value = MODID)
+	public static final Enchantment smelt = null;
+	public GlobalLootModifiersTest() {
+		if (ENABLE)
+		{
+			LootModifierManager.registerFunction(new SmeltingEnchantmentModifier.Serializer());
+			LootModifierManager.registerFunction(new SilkTouchTestModifier.Serializer());
+			LootModifierManager.registerFunction(new WheatSeedsConverterModifier.Serializer());
+		}
+	}
+	
+	@EventBusSubscriber(modid = MODID, bus = EventBusSubscriber.Bus.MOD)
+	public static class EventHandlers {
+		@SubscribeEvent
+		public static void registerEnchantments(@Nonnull final RegistryEvent.Register<Enchantment> event) {
+			event.getRegistry().register(new SmelterEnchantment(Rarity.UNCOMMON, EnchantmentType.DIGGER, new EquipmentSlotType[] {EquipmentSlotType.MAINHAND}).setRegistryName(new ResourceLocation(MODID,"smelt")));
+		}
+	}
+	
+	private static class SmelterEnchantment extends Enchantment {
+		protected SmelterEnchantment(Rarity rarityIn, EnchantmentType typeIn, EquipmentSlotType[] slots) {
+			super(rarityIn, typeIn, slots);
+		}
+	}
+
+	/**
+	 * The smelting enchantment causes this modifier to be invoked, via the smelting loot_modifier json
+	 * @author Draco18s
+	 *
+	 */
+	private static class SmeltingEnchantmentModifier extends LootModifier {
+		public SmeltingEnchantmentModifier(ResourceLocation name, ILootCondition[] conditionsIn) {
+			super(name, conditionsIn);
+
+		}
+
+		@Override
+		public List<ItemStack> doApply(List<ItemStack> generatedLoot, LootContext context) {
+			ArrayList<ItemStack> ret = new ArrayList<ItemStack>();
+			generatedLoot.forEach((stack) -> ret.add(smelt(stack, context)));
+			return ret;
+		}
+
+		private static ItemStack smelt(ItemStack stack, LootContext context) {
```

Comment:
you can use lambdas to make the code cleaner
https://gist.github.com/loordgek/34dde5c4682b7bec38f80db03024183d

Reply:
It's just the test mod, it's not part of the code that is intended to be shipped to modders. But thanks, I'll update with it anyway.